docs/config/gson_config.md

```diff
@@ -0,0 +1,297 @@
+                    new InputStreamReader(new FileInputStream(file), StandardCharsets.UTF_8)
+            )) {
+                // Parses the config file and puts the values into config object
+                config = gson.fromJson(fileReader, MyModConfig.class);
+            } catch (IOException e) {
+                throw new RuntimeException("[MyMod] Problem occurred when trying to load config: ", e);
+            }
+        }
+        // gson.fromJson() can return null if file is empty
+        if (config == null) {
+            config = new MyModConfig();
+        }
+        
+        // Saves the file in order to write new fields if they were added
+        config.saveConfigFile(file);
+        return config;
+    }
+
+    /**
+     * Saves the config to the given file.
+     *
+     * @param file file to save config to
+     */
+    public void saveConfigFile(File file) {
+        try (Writer writer = new OutputStreamWriter(new FileOutputStream(file), StandardCharsets.UTF_8)) {
+            gson.toJson(this, writer);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+}
+```
+
+Now that we have those new methods, we should use the following code to load the config.
+```java
+File configFile = FabricLoader.getInstance().getConfigDir().resolve("my_mod_config.json");
+MyModConfig config = MyModConfig.loadConfigFile(configFile);
+
+if(config.show)
+    System.out.println(config.message);
+
+config.show = false;
+
+// We can save it now
+// If you'd like, you can also save the file parameter when loading
+config.save(configFile);
+
+// Do other stuff with values ...
+```
+
+This will automatically create new config file or read the existing one.
+
+
+## Nested classes
+
+Creating nested structures is really simple. All you need to have is another
+static inner class and a new object for it. See the example below.
+
+```java
+public class MyModConfig {
+
+    public float activationRange = 8.0F;
+
+    public boolean show = true;
+
+    public String message = "This is a config guide.";
+    
+    public NestedValues nested = new NestedValues();
+    
+    public static class NestedValues {
+        public String message = "This is a another message.";
+    }
+    
+}
+```
+
+And its json counterpart:
+
+```json
+{
+    "activationRange": 8.0,
+    "show": true,
+    "message": "This is a config guide.",
+    "nested": {
+        "message": "This is a another message."
+    }
+}
+```
+
+
+## Arrays
+
+To create a json array, you can use `ArrayList<>` class from java.
+
+```java
+public class MyModConfig {
+
+    public float activationRange = 8.0F;
+
+    public boolean show = true;
+
+    public String message = "This is a config guide.";
+    
+    public NestedValues nested = new NestedValues();
+    
+    public static class NestedValues {
+        public String message = "This is a another message.";
+    }
+    
+    public ArrayList<String> randomQuestions = new ArrayList<>(Arrays.asList(
```

Comment:
```suggestion
    public List<String> randomQuestions = new ArrayList<>(Arrays.asList(
```
Usually when storing lists in a field you don't care about the implementation type

Reply:
Thanks, wasn't aware of that :)